gnovm/pkg/gnolang/realm.go

```diff
@@ -167,7 +189,7 @@ func (rlm *Realm) String() string {
 // if rlm or po is nil, do nothing.
 // xo or co is nil if the element value is undefined or has no
 // associated object.
-func (rlm *Realm) DidUpdate(po, xo, co Object) {
+func (rlm *Realm) DidUpdate(po, xo, co Object, store Store) {
```

Comment:
```suggestion
func (rlm *Realm) DidUpdate(po, xo, co Object) {
```

I believe we shouldn't compute the "private" flag at DidUpdate. Instead, we should do it when we lazily load a dependency, once per realm. Then, we can simply read the cached value without accessing the store here.

Reply:
My plan is to: 
- Add private field to objectID & since on setting ID we have access to realm set the object as private, this way we does not need access to store at all or do a map of pkgPath
- add the check in finalize realm tx instead of DidUpdate
Will do this tomorrow morning, but it should start looking solid with this IMO

Reply:
What do you think of this approach, especially this file: https://github.com/gnolang/gno/pull/4422/files#diff-70fe2715458fe8cdaa491a7a2fdd4607c405163c65e80bb11695509e9aaeb9c0